src/sources/amqp.rs

```diff
@@ -213,22 +214,28 @@ fn populate_event(
     );
 
     log_namespace.insert_source_metadata(
-        "amqp",
+        AmqpSourceConfig::NAME,
         log,
-        keys.routing_key_field,
+        Some(LegacyKey::InsertIfEmpty(keys.routing_key_field)),
         "routing",
```

Comment:
Should these theoretically be `path!`?

Reply:
No. These should be parsed as potentially nested paths. The config for `routing_key_field` will be switched to `OwnedTargetPath` in the near future, so it won't be using "String" paths then.

Reply:
Wouldn't that make [this](https://github.com/vectordotdev/vector/blob/master/src/sources/aws_s3/sqs.rs#L527) wrong?

Reply:
`path!("bucket")` is a single segment path, so the macro handles that correctly. If the string literal was a path with multiple segments (eg: `bucket.foo`) than just passing the whole thing to the `path` macro would be incorrect (`path!("bucket.fo")`, instead you would need to manually split it and call it like `path!("bucket", "foo")`

Reply:
Ah, yeah because this is user submitted - we can't be sure. Thanks!